addon/model.js

```diff
@@ -272,6 +272,15 @@ export default class MegamorphicModel extends EmberObject {
   }
 
   notifyPropertyChange(key) {
+    if (CUSTOM_MODEL_CLASS) {
+      // just super and move on for state flags
+      // this needs to match whatever we are notifying
+      // in our subscription to the notificationManager
+      if (['isNew', 'isDeleted'].indexOf(key) !== -1) {
+        super.notifyPropertyChange(key);
+        return;
+      }
```

Comment:
Why are these the only state flags we need to exit early for? do we not have a similar issue with `record.reload()` and `isLoading` / `isLoaded`?

Reply:
Currently ember-data notifies property change for `isNew` and `isDeleted`. Loading doesn't get notified, it is computed by us. We might want to expand this preemptively though

Reply:
Specifically EmberData notifies state change, ember-M3 chooses to notify only these flags on state change. I could believe it needs to notify more things.